pkg/backend/filestate/store.go

```diff
@@ -232,6 +232,22 @@ func (p *projectReferenceStore) ListProjects(ctx context.Context) ([]tokens.Name
 	return projects, nil
 }
 
+func (p *projectReferenceStore) ProjectExists(ctx context.Context, projectName string) (bool, error) {
+	if projectName == "" {
+		return false, fmt.Errorf("empty project name")
+	}
```

Comment:
I think this would be incorrect use of this API. Its contract is that projectName must be specified.
We can use `contract` here to enforce that.

```suggestion
	contract.Requiref(projectName != "", "projectName", "must not be empty")
```

Reply:
I see. I just noticed about the use of the `contract` pkg.

I will do the update.

Reply:
Updated https://github.com/pulumi/pulumi/pull/12798/commits/48756f68b1ec77b93f1ada9bc9dba98a5d1ad806